packages/base/adapters/solana/web3js/client.ts

```diff
@@ -220,21 +230,19 @@ export class SolanaWeb3JsClient implements ChainAdapter<SolStoreUtilState, CaipN
       ...clientOptions.solanaConfig.auth
     })
 
-    this.syncRequestedNetworks(chains, this.options?.chainImages)
-
-    const chain = SolHelpersUtil.getChainFromCaip(
-      chains,
-      SafeLocalStorage.getItem(SolConstantsUtil.CAIP_CHAIN_ID) || ''
-    )
+    const defaultChain = this.defaultChain
 
-    this.defaultChain = chain as CaipNetwork
-    this.syncRequestedNetworks(chains, this.options?.chainImages)
+    if (this.defaultSolanaChain) {
+      SolStoreUtil.setCurrentChain(this.defaultSolanaChain)
+      SolStoreUtil.setCaipChainId(`solana:${this.defaultSolanaChain.chainId}`)
```

Comment:
The `defaultChain` that users pass to the `createWeb3Modal` is not exist and this is handled by us with the v6. That's the reason I've keep the `options.defaultChain` to `this.defaultSolanaChain` since it's type definition is not mathcing with the `CaipNetwork`.

`CaipNetwork` is the type that we want to use globally on every adapter and we are casting the other chain object to this. But in v5 we don't have this. So `this.defaultSolanaChain.id` is not available.